### PR TITLE
[flutter_tools] remove several AndroidSdk Mocks

### PR DESCRIPTION
**Pull request by jonahwilliams**
Work towards https://github.com/flutter/flutter/issues/71511

### CODE REVIEW

File: packages/flutter_tools/test/commands.shard/permeable/build_appbundle_test.dart [hunk 1]
```diff
@@ -97,64 +96,15 @@ void main() {
...
-    MockAndroidSdk mockAndroidSdk;
-    String gradlew;
+    FakeProcessManager processManager;
+    FakeAndroidSdk mockAndroidSdk;
     TestUsage testUsage;
 
     setUp(() {
       testUsage = TestUsage();
       tempDir = globals.fs.systemTempDirectory.createTempSync('flutter_tools_packages_test.');
-      gradlew = globals.fs.path.join(tempDir.path, 'flutter_project', 'android',
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/commands.shard/permeable/build_appbundle_test.dart:108]: There were no verify calls or anything, so the test doesn't really depend on this mock at all. Switched to .any and everything still passes

File: packages/flutter_tools/test/general.shard/android/android_device_stop_test.dart [hunk 2]
```diff
@@ -28,4 +28,4 @@ void main() {
   });
 }
 
-class MockAndroidSdk extends Mock implements AndroidSdk {}
+class FakeAndroidSdk extends Fake implements AndroidSdk {}
```
Comment (jmagman) [inline packages/flutter_tools/test/general.shard/android/android_device_stop_test.dart:31]: Surprised I didn't catch this one.

File: packages/flutter_tools/test/general.shard/android/gradle_test.dart [hunk 3]
```diff
@@ -1031,6 +1030,6 @@ class FakeGradleUtils extends GradleUtils {
   }
 }
 
-class MockAndroidSdk extends Mock implements AndroidSdk {}
+class FakeAndroidSdk extends Fake implements AndroidSdk {}
```
Comment (jmagman) [inline packages/flutter_tools/test/general.shard/android/gradle_test.dart:1033]: Or this.  Gotta up my game.

**Review by jmagman — APPROVED** [review]
LGTM

**Review by leha081 — APPROVED** [review]

**Pull request merged**